### PR TITLE
feat: expose additional tx fields

### PR DESCRIPTION
**Pull request by zivkovicmilos**
## Description

This PR exposes additional TxResult fields to the graphQL interface:
- response info
- response log
- response data
- response error

### CODE REVIEW

File: serve/graph/model/transaction.go
Comment (ajnavarro) [inline serve/graph/model/transaction.go:68]: WDYT about having a Response GraphQL object?

File: serve/graph/schema/types/transaction.graphql
Comment (ajnavarro) [inline serve/graph/schema/types/transaction.graphql:38]: The same, we can have here a Response object.

File: serve/graph/transaction.resolvers.go
Comment (ajnavarro) [inline serve/graph/transaction.resolvers.go:16]: the graphql framework generated this because types are not matching. You are returning []byte from the model, but requesting !String from the schema.

File: serve/graph/model/transaction.go
Comment (jinoosss) [reply serve/graph/model/transaction.go:68]: I agree with @ajnavarro .

How about a Response GraphQL object like this ?

```
"""
`TransactionResponse` is the processing result of the transaction.
It has `log`, `info`, `error`, and `data`.
"""
type TransactionResponse {
  """
  The log value associated with the Transaction execution, if any.
  """
  log: String!

  """
  The Info associated with the Transaction execution, if any.
  """
  info: String!

  """
  The error value associated with the Transaction execution, if any.
  """
  error: String!

  """
  The response data associated with the Transaction execution, if any.
  """
  data: String!
}
```

Comment (jinoosss) [reply serve/graph/model/transaction.go]: Can you please review the implementation ?

https://github.com/gnolang/tx-indexer/pull/32/commits/c0f0cf3c964a4e1bcd0738cfd7254a010329a9cc#diff-3a6df2b898459c1183102e35be08aa58206aadf7c363a393dae0db228d781c1bR126-R149

File: serve/graph/transaction.resolvers.go
Comment (jinoosss) [reply serve/graph/transaction.resolvers.go]: I fixed it in this commit. 
https://github.com/gnolang/tx-indexer/pull/32/commits/8cf8f5ccae08afa56bbde31f570356ff3b4a23cb

**Comment by zivkovicmilos** [comment]
@jinoosss can you please update this PR with the latest `main` changes? 🙏

**Comment by zivkovicmilos** [comment]
I think we should be good to go, just waiting on @ajnavarro to give the final approve for the resolved comments

File: serve/graph/transaction.resolvers.go
Comment (ajnavarro) [inline serve/graph/transaction.resolvers.go:22]: This is not needed. Add a Response() method to Transaction Model

**Review by ajnavarro — CHANGES_REQUESTED** [review]

File: serve/graph/transaction.resolvers.go
Comment (jinoosss) [reply serve/graph/transaction.resolvers.go]: Thank you. 
I didn't see any code changes with the addition of the merge commit.

File: serve/graph/model/transaction.go
Comment (ajnavarro) [inline serve/graph/model/transaction.go:56]:
```suggestion
func (t *Transaction) Response() *TransactionResponse {
	return &TransactionResponse{response: t.txResult.Response}
}
```

**Review by ajnavarro — CHANGES_REQUESTED** [review]
Just a minor change and after that we can merge!

**Review by ajnavarro — APPROVED** [review]

**Pull request merged**
